src/server/cray/cfs/api/dbutils/db_wrapper.py

```diff
@@ -277,24 +281,189 @@ def delete(self, key: DbKey) -> None:
         """
         self.get_delete(key)
 
+    def _delete_batch(self,
+        keys: Iterable[str],
+        data_filter: DataFilter,
+        keys_done: set[str]
+    ) -> tuple[list[str], list[DbEntry]]:
+        """
+        Helper for delete_all method
+        Given a list of keys, tries to delete all of the keys
+        which exist in the database and which meet the specified
+        filter (if one is specified).
+
+        Before attempting the delete, as keys are excluded based on
+        the above criteria, they are added to 'keys_done'.
+
+        If the delete fails because of database changes that occur,
+        then a Redis WatchError will be raised, which the caller must
+        handle as they see fit.
+
+        If the delete succeeds, this function returns two lists:
+        1. A list of database keys that were deleted
+        2. The data entries for each key in #1
+
+        If no keys meet the criteria for deletion, then this function
+        will return two empty lists.
+        """
+        keys_to_delete: list[str] = []
+        data_to_delete: list[DbEntry] = []
+        with self.client.pipeline() as pipe:
+            # Start by watching all of the keys in this batch
+            # This has to be done before we call mget for them.
+            pipe.watch(*keys)
+
+            # Retrieve all of the specified keys from the database
+            # All database calls to pipe will be executed immediately,
+            # since we have not yet called pipe.multi()
+            data_str_list = pipe.mget(*keys)
+
+            for key, data_str in zip(keys, data_str_list):
+                if not data_str:
+                    # Already empty, no need to delete it
+                    keys_done.add(key)
+                    continue
+                data = json.loads(data_str)
+                if data is None:
+                    keys_done.add(key)
+                    continue
```

Comment:
Remove `data is None` check per [comment in other PR](https://github.com/Cray-HPE/config-framework-service/pull/206/files#r2437237282)
```suggestion
```